x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -51,3 +68,47 @@ export const transformValidateBulkError = (
     });
   }
 };
+
+// Temporary functionality until the new System Actions is in place
+// for now we want to make sure that user cannot configure Isolate action if has no RBAC permissions to do so
+export const validateResponseActionsPermissions = async (
+  securitySolution: SecuritySolutionApiRequestHandlerContext,
+  ruleUpdate: RuleCreateProps | RuleUpdateProps,
+  existingRule?: RuleAlertType | null
+): Promise<void> => {
+  const { experimentalFeatures } = await securitySolution.getConfig();
+
+  if (!experimentalFeatures.endpointResponseActionsEnabled || !isQueryRule(ruleUpdate.type)) {
+    return;
+  }
+
+  const payload = ruleUpdate as QueryRule;
+  const existingPayload = existingRule as Rule<UnifiedQueryRuleParams>;
+
+  if (payload.response_actions?.length || existingPayload?.params?.responseActions?.length) {
+    const endpointAuthz = await securitySolution.getEndpointAuthz();
+
+    const difference = findDifferenceInArrays<ResponseAction, RuleResponseAction>(
```

Comment:
Well, I had to support empty arrays, because it was not finding it as differences. However now the tests are still passing when I removed those So... I will have to give it another try and test it. :) Thanks!

Reply:
Yes I tested and it doesn't find differences correctly when we eg. had 1 response action in existingRule, and remove it to end up with empty array. I added a test to validate that.

Reply:
Good point! I've checked the [implementation](https://github.com/lodash/lodash/blob/master/.internal/baseDifference.js#L24) in lodash and it seems considering comparing to an empty array as there are no differences. Looks like a bug IMHO.